collect_app/src/main/java/org/odk/collect/android/preferences/LocaleHelper.java

```diff
@@ -0,0 +1,60 @@
+package org.odk.collect.android.preferences;
+
+import android.content.Context;
+import android.content.res.Configuration;
+import android.os.Build;
+import android.preference.PreferenceManager;
+import android.util.DisplayMetrics;
+
+import java.util.Locale;
+import java.util.TreeMap;
+
+/**
+ * Changes the locale of the app and keeps the changes persistent
+ *
+ * @author abdulwd
+ */
+
+public class LocaleHelper {
+
+    public void updateLocale(Context context) {
+        String localeCode = PreferenceManager.getDefaultSharedPreferences(context)
+                .getString(PreferenceKeys.KEY_LANGUAGE, "en");
+        Locale locale = getLocale(localeCode);
+        Locale.setDefault(locale);
+        Configuration configuration = new Configuration();
+        DisplayMetrics displayMetrics = context.getResources().getDisplayMetrics();
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            configuration.setLocale(locale);
+        } else {
+            configuration.locale = locale;
+        }
+        context.getResources().updateConfiguration(configuration, displayMetrics);
+        context.getApplicationContext().getResources().updateConfiguration(configuration, displayMetrics);
+    }
+
+    public TreeMap<String, String> getEntryListValues() {
+        String[] languages = {"af", "am", "ar", "bn", "ca", "cs", "de", "en",
```

Comment:
How about making this a constant in `utilities.ApplicationConstants` called something like `TRANSLATIONS_AVAILABLE`.